src/NeuralNet/Initializers/Base/Contracts/AbstractInitializer.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rubix\ML\NeuralNet\Initializers\Base\Contracts;
+
+use Rubix\ML\NeuralNet\Initializers\Base\Exceptions\InvalidFanInException;
+use Rubix\ML\NeuralNet\Initializers\Base\Exceptions\InvalidFanOutException;
+
+/**
+ * Abstract Initializer for init params validation
+ *
+ * @category    Machine Learning
+ * @package     Rubix/ML
+ * @author      Andrew DalPino
+ * @author      Aleksei Nechaev <omfg.rus@gmail.com>
+ */
+abstract class AbstractInitializer implements Initializer
+{
+    /**
+     * Validating initializer parameters
+     *
+     * @param int $fanIn The number of input connections per neuron
+     * @param int $fanOut The number of output connections per neuron
+     * @throws InvalidFanInException Initializer parameter fanIn is less than 1
+     * @throws InvalidFanOutException Initializer parameter fanOut is less than 1
+     */
+    protected function validateInitParams(int $fanIn, int $fanOut) : void
+    {
+        if ($fanIn < 1) {
+            throw new InvalidFanInException(message: "Fan in cannot be less than 1, $fanIn given");
```

Comment:
To me, this exception does not meet the threshold for having it's own class. I would roughly define that threshold as being ...

1. Will the user want to catch this exception specifically? - IMO probably not
2. Does the exception have complex logic? - It does not